theano/gof/utils.py

```diff
@@ -546,28 +546,28 @@ def remove(predicate, coll):
     import hashlib
 
     def hash_from_code(msg):
-        # hashlib.md5() requires an object that supports buffer interface,
+        # hashlib.sha256() requires an object that supports buffer interface,
         # but Python 3 (unicode) strings don't.
         if isinstance(msg, str):
             msg = msg.encode()
         # Python 3 does not like module names that start with
         # a digit.
-        return 'm' + hashlib.md5(msg).hexdigest()
+        return 'm' + hashlib.sha256(msg).hexdigest()
 
 else:
     import hashlib
 
     def hash_from_code(msg):
         try:
-            return hashlib.md5(msg).hexdigest()
+            return hashlib.sha256(msg).hexdigest()
         except TypeError:
             assert isinstance(msg, np.ndarray)
```

Comment:
Under which circumstance would a code string be a numpy array?  This doesn't make sense.  It also appear that this can't be the case under python 3 for some reason.

I would unify the two functions since I'm sure that python2 will work with the python 3 code if we swap `str` for `unicode`